RRM: getPublications selector

ankitrox opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- #8794 

## Relevant technical choices

<!-- Please describe your changes. -->

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

nfmohit commented on assets/js/modules/reader-revenue-manager/datastore/publications.test.js:
```diff
@@ -0,0 +1,123 @@
+					status: 200,
+				} );
+
+				const initialProperties = registry
```
The variable name here appears wrong.

```suggestion
				const initialPublications = registry
```

nfmohit commented on assets/js/modules/reader-revenue-manager/datastore/publications.test.js:
```diff
@@ -0,0 +1,123 @@
+				);
+			} );
+
+			it( 'should not make a network request if properties for this account are already present', async () => {
```
```suggestion
			it( 'should not make a network request if publications are already present', async () => {
```

nfmohit commented on assets/js/modules/reader-revenue-manager/datastore/__fixtures__/publications.json:
While we're at it, it'd be nice to have a few more publication fixtures with different `onboardingState` values, e.g. `ONBOARDING_ACTION_REQUIRED` and `ONBOARDING_COMPLETE`.

nfmohit replied:
It looks like a publication with `ONBOARDING_COMPLETE` was not added, and one with `ONBOARDING_STATE_UNSPECIFIED` was added twice. Let's change the other one to have `ONBOARDING_COMPLETE`.

nfmohit reviewed (changes requested):
Excellent work on this, thank you @ankitrox ! I've left a few comments, after addressing which, we should be able to get this merged.

Also, could you please update the QAB and mention activating the `rrmModule` feature flag and the Reader Revenue Manager module, otherwise, the request won't work.

Please take a look and feel free to let me know if you have any questions. Thank you!

ankitrox commented:
> Excellent work on this, thank you @ankitrox ! I've left a few comments, after addressing which, we should be able to get this merged.
> 
> Also, could you please update the QAB and mention the following:
> 
> 1. Activating the `rrmModule` feature flag and the Reader Revenue Manager module, otherwise, the request won't work.
> 2. It is not guaranteed that the response will be empty. The Google account used to test this may actually have publications.
> 
> Please take a look and feel free to let me know if you have any questions. Thank you!

Thanks @nfmohit .

I've updated the QAB and addressed the feedback.

Over to you for re-review.

nfmohit reviewed (changes requested):
Thank you for addressing my feedback, @ankitrox ! Please take a look at the following additional points:

1. I have left two very minor comments below.
2. We need to update the PR branch my merging `develop` into it and fixing conflicts.
3. I should've mentioned this earlier, but the QAB should be updated to mention that the response will only include publications that are associated with the site URL.

As the above feedback is quite minor, I'll go ahead and apply them myself, I hope that's okay, thank you!

nfmohit reviewed (approved):
LGTM 👍
